Plugin/Catalog/Block/Toolbar.php

```diff
@@ -191,10 +206,18 @@ private function getCmpResult(Store $store, FulltextCollection $collection)
         $nostoCustomer = $this->cookieManager->getCookie(NostoCustomer::COOKIE_NAME);
         $limit = $collection->getSize();
         $personalizationResult = null;
+
+        // Build filters
+        $this->nostoFilterBuilder->init($store);
+        $this->nostoFilterBuilder->buildFromSelectedFilters(
+            $this->state->getActiveFilters()
+        );
+
         ServerTiming::getInstance()->instrument(
             function () use ($nostoAccount, $nostoCustomer, $categoryString, $limit, &$personalizationResult) {
```

Comment:
This looks messy. I would move these https://github.com/Nosto/nosto-magento2-cmp/blob/7f5807e0d20690d5a67327e6470fd5dcc4a79630/Plugin/Catalog/Block/Toolbar.php#L204-L206 here https://github.com/Nosto/nosto-magento2-cmp/blob/7f5807e0d20690d5a67327e6470fd5dcc4a79630/Plugin/Catalog/Block/Toolbar.php#L219-L222